File: core/src/rustyice/game/GameObject.java
package rustyice.game;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import rustyice.graphics.Camera;
import rustyice.graphics.RenderFlags;

import java.util.EnumSet;

public interface GameObject {

    void init();

    void store();

    void update(float delta);

    void render(SpriteBatch batch, Camera camera, int flags);

    boolean isInitialized();

    float getX();

    float getY();

    Section getSection();
}

File: core/src/rustyice/game/GameObject.kt
package rustyice.game

abstract class GameObject: GameLifecycle(){

    abstract val section: Section?
    val game: Game? get() = section?.game

    abstract var x: Float
    abstract var y: Float
    abstract var rotation: Float

    fun setPosition(x: Float, y: Float) {
        this.x = x
        this.y = y
    }
}
